### ISSUE TITLE
Run all tests on release candidate [Chrome]

### ISSUE THREAD

**Issue by dhowe** [dhowe/AdNauseam#572]
First, complete ticket #571

Then, in your browser of choice, install the tag (via git) or release from the [releases page](https://github.com/dhowe/AdNauseam/releases) and follow the [testing instructions](https://github.com/dhowe/AdNauseam/wiki/Testing-New-Release-Candidates) to test the release as per usual. Note that you can ignore all tests relating to DNT, as it is on hold until next version...

Add whatever issues you find as new tickets, then list them by (linked) number below.

**Comment by leoneckert** [comment]
**Chrome**

- [ ] Interface
- [ ] First-Run
- [ ] Menu 
- [ ] Vault
- [ ] Notifications
- [ ] Ad Hiding/Parsing
- [ ] Dashboard
- [ ] DoNotTrack (DNT)
- [ ] Cookies 
- [ ] Ad Visits 
- [ ] Tools

**Comment by dhowe** [comment]
Please create new tickets for the items above (with clear steps on how to recreate the issue), and then add the links to the tickets in the list above

**Comment by dhowe** [comment]
bing text-ads seems ok for me (in any case, create a ticket to retest these):

![screen shot 2016-11-05 at 1 20 41 am](https://cloud.githubusercontent.com/assets/737638/20015357/20c0d626-a2f6-11e6-8524-f2cc4852f056.png)

**Comment by leoneckert** [comment]
tickets created for issues found above. 
(additional comments: in search engines, only bing.com still gives me ads (https://github.com/dhowe/AdNauseam/issues/578) // the zoom 'issue' in vault I mentioned above, comes up 1 on 50 times, when zooming in and out with click wheel and zoom buttons in exchange. There is no precise recipe to re-create)

now looking into the cookies again.

**Comment by leoneckert** [comment]
hey @dhowe , do i understand this right now? 
my settings are these:
<img width="363" alt="screen shot 2016-11-04 at 18 28 51" src="https://cloud.githubusercontent.com/assets/14130931/20024373/987d5bba-a2bc-11e6-9faa-b50dd3126c4f.png">

the ad I am collecting has this target:
<img width="899" alt="screen shot 2016-11-04 at 18 28 44" src="https://cloud.githubusercontent.com/assets/14130931/20024386/a4a8cda2-a2bc-11e6-91d4-de82a51299a0.png">

and the same url in the requests I am watching has this header:
<img width="1214" alt="screen shot 2016-11-04 at 18 30 36" src="https://cloud.githubusercontent.com/assets/14130931/20024422/de00340a-a2bc-11e6-864c-9320bd01b040.png">


-> I assume that means, it is 
[+] positive that indeed no cookie is sent (there is one in the response, but I assume that is fine as long as we don't send one), but 
[-] negative because there is the referrer (nytimes.com) being sent in the request.

**Comment by dhowe** [comment]
Yes, there MUST be a request made to the original ad-target. You should see this in the AdNauseam console. For example:

![screen shot 2016-11-06 at 3 12 35 pm](https://cloud.githubusercontent.com/assets/737638/20035943/7be17068-a433-11e6-9907-50ad00b45c6c.png)

If the referrer is being sent when 'hide referrer' is checked, then this is a bug, and you should create a new ticket for it.

You should first try this with a very simple page with a single ad, [like this](), then once everything works OK, and you understand and can view the requests, then test with real pages.

**Comment by dhowe** [comment]
Lets move the above to a separate ticket, so that Leon can continue his tests here

**Comment by dhowe** [comment]
@leoneckert the referer problem you mention above is quite perplexing as the the referrer for an XMLHttpRequest (which we use for all of our visits) does not include a referer unless we explicitly add one. So if this is the case, then there is a big bug in our code (one that I cannot recreate)... You need to create a very simple test case for such things (using http://rednoise.org/adntest/simple.html for example)

**Comment by leoneckert** [comment]
(back on this now again) update to follow

**Comment by leoneckert** [comment]
so, I am pretty sure I have finally a clear understanding of the cookies. **I can confirm that no cookies from websites that are 'visited' by AdN are accepted/stored.** 
To analyse I made sure I am working on a all blank situation (clear browser profile, fresh ADN install, open background console), then visited one single site and analysed the cookie-situation.

In the Networks tab(s), for every request, the response header can be seen which often has the "set-cookie" header. Cookies that came as responses from sites visited by ADN (that is "set-cookie"headers found in the background.html console) DO NOT match with the cookies that end up in the local Chrome storage. 
Cookies in the responses of requests of the actual site DO MATCH up (if that's what the browser's setting tell it to do). 

-> I conclude, ADN-caused cookies are not accepted. All other cookies are dealt with according to the user's browser settings. 

To illustrate my inspection and understanding of this, please check if this graphic makes sense ( @dhowe ) :

![cookie_analysis](https://cloud.githubusercontent.com/assets/14130931/20201117/aa5b407e-a783-11e6-84ff-9c938e70c3ee.jpg)

**Comment by leoneckert** [comment]
- [X] "adVisits" of tests

**Comment by leoneckert** [comment]
with that, the chrome tests should be all done.

**Comment by dhowe** [comment]
I'm a little unclear on the cookie scenario above. First, you are correct that the `set-cookie` header will show for response to our XHR clicks in the network panel; the response does in fact have these headers, but we do not allow the cookies to actually be set. This is expected

The images above are a bit difficult to read, but are you suggesting that cookies with the same name but different data are showing up in the user's cookie panel? 

This _could_ be correct, assuming the following:
- A) the user is allowing 3rd-party cookies in the browser settings 
- B) the 3rd party Ad was not ALLOWed from an active block list, and
- C) the 3rd party domain from which we request the ad image is the same as the domain in the Ad's targetURL (e.g., doubleclick is both hosting the image, and receiving our click, at least before any redirect). 

So we need to test each of these cases separately...  A) is relatively simple (no 3rd part cookies should ever show up when disabled by the browser), but B/C likely require a simple test page(s) to be created, with just one ad that is served from a request that is on an active list, but ALLOWed by AdNauseam, to a site that attempts to set a cookie. We need to do this once if the image and target domains are the same, and once if different.

For the moment, @leoneckert, please just read the above and confirm that we both agree on the above, then we will do the testing post 2.4...

**Comment by leoneckert** [comment]
@dhowe read and agreed, in the case above, the browser settings did allow 3rd-party-cookies as this:
<img width="636" alt="screen shot 2016-11-12 at 13 10 21" src="https://cloud.githubusercontent.com/assets/14130931/20239897/62e937d0-a8d9-11e6-82e3-56bbddfd3c71.png">

I am not quite sure what you mean by a request that IS ON an active list, but ALLOWed. Isn't it not ALLOWed when on the list? 

will spend further time on this post 2.4

**Comment by dhowe** [comment]
@leoneckert read this: https://github.com/dhowe/AdNauseam/wiki/Developer-FAQ#what-is-the-relationship-between-blocking-and-hiding-rules-in-ublock-and-adn
and see if it clears things up  ?

**Comment by leoneckert** [comment]
@dhowe yes it does, thanks!

**Comment by leoneckert** [comment]
**Chrome**

- [x] Interface
    - [x] First-Run
    - [x] Menu
        Notes:
            - uBlock is version 1.9.7.100 (not 1.9.6)
            - zooming back after zooming, small cosmetic z-index error
    - [x] Vault
        Notes:
            - when inspector is active and window is resized, the inspected ad disappears with the background remaining black (lightbox mode)
            - one thing about the date slider: I don't find it intuitive to figure out that on the edges of it on can grab and slide. Once that is done it's super intuitive (dragging the box or even using the little cross to mark a new box), but the first step I would have never found if I wasn't working on this project. Maybe there should be a indication that the slider can be shrunk to select specific periods?
    - [x] Notifications:
        Notes:
            - No notifications in menu when on Paused page (e.g. Dashboard), but I doubt this is not wanted - just noticed it for the first time right now when expecting to see them. NO ISSUE.

- [x] Ad Hiding/Parsing
        Notes:
            - Are '(cosmetically filtered iframes)[http://rednoise.org/adntest/iframe-cosm.html]' meant to be collected, or only hidden (the latter is the case).
            - The .crx version actually not testable in Incognito mode, because the Incognito window is treated as opening  a *new* chrome and The Extension will then be disabled by Chrome.
- [x] Dashboard
    Notes:
        - uBlock is version 1.9.7.100 (not 1.9.6) (same note as above)
        - the (i) behind the malware option is still not linked
- [ ] Cookies
- [ ] Ad Visits


remaining two test scheduled tomorrow

**Comment by dhowe** [comment]
.crx issues is noted -- please create a separate ticket for any other issues or questions...

**Comment by leoneckert** [comment]
Testing done, going on to other tickets

**Comment by dhowe** [comment]
👍
